lib/application.js

```diff
@@ -617,6 +622,63 @@ app.listen = function listen() {
   return server.listen.apply(server, arguments);
 };
 
+/*
+ * Set a tracer at the application level. Tracer will be activated when a
+ * tracing call will be fired by a response.
+ * The tracer function takes an option object as argument with the given
+ * attributes:
+ *
+ * - app: current application
+ * - res: response that fires the tracer
+ * - req: request related to above response
+ * - event: string describing the event to trace
+ * - date: when the tracing occurs
+ * - args: allow to freely add information
+ *
+ * @param {function} The tracer to set.
+ * @return {app} for chaining.
+ * @public
+ */
+
+app.instrument = function(tracer){
+  if (tracer === undefined || typeof tracer !== 'function')
+    throw new TypeError('instrument expects a function');
+  this.tracers.push(tracer);
+  return this;
+};
+
+/*
+ * Call all tracers set on the application. Add context information:
+ * running app, request, response and date. The date is the when the tracing
+ * was fired (now).
+ *
+ * @param {Response} Response that fires the tracing event.
+ * @param {Event} The event to trace.
+ * @param {Array} Arguments to transmit to the tracker.
+ *
+ * @private
+ */
+
+app._callTracers = function(res, event, args){
+  var date = new Date();
+  var app = this;
+  var tracers = this.tracers;
+  var length = tracers.length;
+
+  var options = {
+    app: app,
+    res: res,
+    req: res.req,
+    event: event,
+    date: date,
+    args: args
+  };
+
+  for (var i = 0; i < length; i++) {
+    tracers[i](options);
```

Comment:
> Do you know what is the best practice in this case?

I'm not sure, because that is a question that has to do with what exactly the usage of `app.instrument` is supposed to be. Something needs to be done, and I was hoping you would know, since you made the PR :) I can help decide, but need to understand a lot better how this PR would be used in practice, etc. You know, the code here is being built for someone to use, and those users' needs are what this needs to cater towards. I don't really have the need, so it's hard for me to answer this question unless you can help educate me on the needs :)